public/app/features/dashboard/components/ShareModal/ShareModal.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { PureComponent } from 'react';
+import { css } from 'emotion';
+import { GrafanaTheme } from '@grafana/data';
+import { stylesFactory, Modal, TabsBar, Tab, TabContent, withTheme, Themeable, Icon } from '@grafana/ui';
+import { DashboardModel, PanelModel } from 'app/features/dashboard/state';
+import { ShareLink } from './ShareLink';
+import { ShareSnapshot } from './ShareSnapshot';
+import { ShareExport } from './ShareExport';
+import { ShareEmbed } from './ShareEmbed';
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+  modalHeaderTitle: css`
+    font-size: ${theme.typography.heading.h3};
+    padding-top: ${theme.spacing.sm};
+    margin: 0 ${theme.spacing.md};
+  `,
+  modalHeaderIcon: css`
+    margin-right: ${theme.spacing.md};
+    font-size: inherit;
+    &:before {
+      vertical-align: baseline;
+    }
+  `,
+}));
+
+const shareModalTabs = [
+  { label: 'Link', value: 'link' },
+  { label: 'Embed', value: 'embed' },
+  { label: 'Snapshot', value: 'snapshot' },
+  { label: 'Export', value: 'export' },
+];
+
+interface Props extends Themeable {
+  isOpen: boolean;
+  dashboard: DashboardModel;
+  panel?: PanelModel;
+
+  onDismiss(): void;
+}
+
+interface State {
+  tab: string;
+}
+
+function getInitialState(): State {
+  return {
+    tab: shareModalTabs[0].value,
+  };
+}
+
+export class ShareModalUnthemed extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = getInitialState();
+  }
+
+  onDismiss = () => {
+    this.setState(getInitialState());
+    this.props.onDismiss();
+  };
+
+  onSelectTab = (t: any) => () => {
+    this.setState({ tab: t.value });
+  };
+
+  renderTabsBar() {
+    const { panel } = this.props;
+    const { tab } = this.state;
+
+    return (
+      <TabsBar hideBorder={true}>
+        {shareModalTabs.map((t, index) => {
+          // Filter tabs for dashboard/panel share modal
+          if ((panel && t.value === 'export') || (!panel && t.value === 'embed')) {
+            return null;
+          }
+
+          return (
+            <Tab
+              key={`${t.value}-${index}`}
+              label={t.label}
+              active={t.value === tab}
+              onChangeTab={this.onSelectTab(t)}
+            />
+          );
+        })}
+      </TabsBar>
+    );
+  }
+
+  renderTitle() {
+    const { theme, panel } = this.props;
+    const styles = getStyles(theme);
+    const title = panel ? 'Share Panel' : 'Share';
+
+    return (
+      <>
+        <h2 className={styles.modalHeaderTitle}>
+          <Icon name="share-square-o" className={styles.modalHeaderIcon} />
+          {title}
+        </h2>
+        {this.renderTabsBar()}
```

Comment:
This should be part of the Modal component somehow, maybe a namespace component on Modal like `Modal.TabsHeader title="asd" icon="share-square-o"><TabsBar>...` So the bascially a Modal.TabsHeader componet that captures the styling you have here for modalHeaderTitle etc .

Reply:
Yes, sounds good

Reply:
Created `ModalTabsHeader` and `ModalHeader` components for reusing modal header. It's a good idea to move it under `Modal` namespace, but I think it's too much work that touches other components, so better do it after merging.